packages/smooth_app/lib/pages/product/new_product_page.dart

```diff
@@ -20,83 +20,60 @@ import 'package:smooth_app/themes/smooth_theme.dart';
 import 'package:smooth_ui_library/widgets/smooth_card.dart';
 
 class NewProductPage extends StatefulWidget {
-  const NewProductPage({
-    required this.product,
-  });
+  const NewProductPage(this.product);
 
   final Product product;
 
   @override
   State<NewProductPage> createState() => _ProductPageState();
 }
 
+enum ProductPageMenuItem { WEB, REFRESH }
+
 class _ProductPageState extends State<NewProductPage> {
   late Product _product;
-  bool _first = true;
 
   @override
   void initState() {
     super.initState();
     _updateHistory(context);
+    _product = widget.product;
   }
 
   @override
   Widget build(BuildContext context) {
     final LocalDatabase localDatabase = context.watch<LocalDatabase>();
     final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
```

Comment:
regarding context.watch vs context.read... don't we want the rebuilding if the product changes in localDatabase ? (although I don't know how it will happen but this was the behavior before).

Regarding retrieving from context, I prefer retrieving in a single place and passing the variable around instead of retrieving in every scope the variable is needed, mostly a matter of taste. Would have been a different story if I was passing a lot of parameters, just passing 2 looks fine to me.

Reply:
thanks, realizing now that `context.read` would work here since we are using `setState()` whenever we update _product. In which case also ok to pass down the context to `_refreshProduct`